components/rmrk/Gallery/Holder/Holder.vue

```diff
@@ -345,37 +366,61 @@ export default class Holder extends mixins(ChainMixin, KeyboardEventsMixin) {
     return Object.values(itemRowMap)
   }
 
-  private generateHolderGroups(itemRowList): TableRow[] {
-    const holderGroups: Record<string, TableRow> = {}
-    itemRowList
-      .filter((item) => item.Holder !== '-')
-      .forEach((item) => {
-        const holder = item['Holder']
-        if (holderGroups[holder]) {
-          holderGroups[holder].Items.push(item)
-          holderGroups[holder]['Bought'] =
-            holderGroups[holder]['Bought'] + item['Bought']
-          holderGroups[holder]['Sale'] =
-            holderGroups[holder]['Sale'] + item['Sale']
-        } else {
-          holderGroups[holder] = {
-            ...item,
-            Items: [item],
-          }
+  private getGroupNameFromRow(item: TableRow): string {
+    if (this.groupKey === 'Holder') {
+      return item['Holder']
+    } else if (this.groupKey === 'CollectionId') {
+      return item['Item']['collection']['id']
+    }
+    return item['Holder']
+  }
+
+  private getCustomRowFilter(): (item: TableRow) => boolean {
+    if (this.groupKey === 'Holder') {
+      return (item) => item.Holder !== '-'
+    } else if (this.groupKey === 'CollectionId') {
+      return (item) => item.Holder === this.$route.params.id
+    }
+    return () => true
+  }
+
+  private generatecustomGroups(itemRowList: TableRow[]): TableRow[] {
+    const customGroups: Record<string, TableRow> = {}
+    itemRowList.filter(this.getCustomRowFilter()).forEach((item: TableRow) => {
+      item = {
+        Bought: 0,
+        Sale: 0,
+        ...item,
+      }
```

Comment:
Ts error here, this is specified more than once + Sale is define as a string but you push number you should change type and do something like:
```ts
item.Bought = item.Sale = 0
```
not sure if this would be a good pratice

Reply:
@Jarsen136 these lines are still problematic, I've seen that you fixed `Sale` type but I still got an error here

Reply:
That is really confusing. Could you please give me the report url, so that I can check where the error come from : )

Reply:
weird fix but ok

Reply:
: ) because the zero value is a fallback value, could not be set to Sate\Bought directly, which will cause an incorrect result.